### PR TITLE
[FEATURE] Added custom routes for product bundle

### PR DESCRIPTION
**Pull request by turbo-ele**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | none
| Related issues/PRs | #337 #339
| License | MIT

#### What's in this PR?

Custom routes for product bundle by using sulu route bundle.

### CODE REVIEW

File: Controller/ProductViewController.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+use Sulu\Bundle\ProductBundle\Entity\ProductInterface;
+use Sulu\Bundle\ProductBundle\Entity\ProductTranslation;
+use Sulu\Bundle\ProductBundle\Product\ProductFactoryInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * This controller is used for viewing product-templates.
+ */
+class ProductViewController extends Controller
```
Comment (wachterjohannes) [inline Controller/ProductViewController.php]: `WebsiteProductController`

File: DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -56,7 +56,7 @@ public function getConfigTreeBuilder()
                     ->defaultValue(['de', 'en'])
                 ->end()
                 ->scalarNode('template')
-                    ->defaultValue('ClientWebsiteBundle:views:product.html.twig')
+                    ->defaultValue('SuluProductBundle:ProductView:product-detail.html.twig')
```
Comment (wachterjohannes) [inline DependencyInjection/Configuration.php]: `SuluProductBundle:WebsiteProduct:product-detail.html.twig`

File: EventListener/ProductTranslationEventListener.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+namespace Sulu\Bundle\ProductBundle\EventListener;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Sulu\Bundle\ProductBundle\Event\ProductTranslationEvent;
+use Sulu\Bundle\ProductBundle\Product\ProductRouteManager;
+
+/**
+ * Event listener for product translation events.
+ */
+class ProductTranslationEventListener
```
Comment (wachterjohannes) [inline EventListener/ProductTranslationEventListener.php]: `ProductTranslationEntityListener`

File: Product/ProductRouteManager.php [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+namespace Sulu\Bundle\ProductBundle\Product;
+
+use Sulu\Bundle\ProductBundle\Entity\ProductTranslation;
+use Sulu\Bundle\RouteBundle\Manager\RouteManagerInterface;
+
+/**
+ * Manager for handling routes of product bundle.
+ */
+class ProductRouteManager
```
Comment (wachterjohannes) [inline Product/ProductRouteManager.php]: interface?

File: Routing/ProductRouteDefaultProvider.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+namespace Sulu\Bundle\ProductBundle\Routing;
+
+use Sulu\Bundle\ProductBundle\Entity\ProductTranslation;
+use Sulu\Bundle\ProductBundle\Entity\ProductTranslationRepository;
+use Sulu\Bundle\RouteBundle\Routing\Defaults\RouteDefaultsProviderInterface;
+
+/**
+ * Provides route-defaults for products.
+ */
+class ProductRouteDefaultProvider implements RouteDefaultsProviderInterface
```
Comment (wachterjohannes) [inline Routing/ProductRouteDefaultProvider.php]: `ProductRouteDefaultsProvider`

File: Controller/ProductViewController.php [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+     * @return Response
+     */
+    public function viewAction(ProductInterface $product, ProductTranslation $translation)
+    {
+        $apiProduct = $this->getProductFactory()->createApiEntity($product, $translation->getLocale());
+
+        return $this->render(
+            $this->getProductViewTemplate(),
+            [
+                'product' => $apiProduct,
```
Comment (wachterjohannes) [inline Controller/ProductViewController.php]: `urls` variable should also be passen which is a assoziative array from locale to url

File: Controller/ProductViewController.php [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+     * This action is used for displaying a product via a template.
+     * The template that is defined by sulu_product.template parameter is used for displaying
+     * the product.
+     *
+     * @param ProductInterface $product
+     * @param ProductTranslation $translation
+     *
+     * @return Response
+     */
+    public function viewAction(ProductInterface $product, ProductTranslation $translation)
```
Comment (wachterjohannes) [inline Controller/ProductViewController.php]: `indexAction`

**Comment by turbo-ele** [comment]
@wachterjohannes Cleanup completed

**Pull request merged**